docker-entrypoint.sh

```diff
@@ -1,19 +1,42 @@
 #!/usr/bin/env sh
-set -eux
+set -eu
+
+set_password() {
+    expect <<-'EOF'
+        set timeout -1
+
+        spawn /usr/local/bin/devpi-passwd root
+
+        expect "enter password" { send "$env(DEVPISERVER_PASSWORD)\r" }
+        expect "repeat password" { send "$env(DEVPISERVER_PASSWORD)\r" }
+
+        expect "committed: keys:"
+        wait
+EOF
+}
 
 if [ -z "${DEVPISERVER_HOST:-}" ]; then
-    DEVPISERVER_HOST="0.0.0.0"
+    export DEVPISERVER_HOST="0.0.0.0"
 fi
 
 if [ -z "${DEVPISERVER_PORT:-}" ]; then
-    DEVPISERVER_PORT="3141"
+    export DEVPISERVER_PORT="3141"
+fi
+
+if [ -z "${DEVPISERVER_PASSWORD:-}" ]; then
+    # Generate a random password if not provided
+    export DEVPISERVER_PASSWORD=$(</dev/urandom tr -dc _A-Z-a-z-0-9 | head -c32)
 fi
 
 if [ ! -f "${DEVPISERVER_SERVERDIR:-}/.nodeinfo" ]; then
     echo "Initializing devpi..."
     devpi-init
```

Comment:
Seems like `devpi-init` accepts root password or hash. Can't we use it instead of the spawn+ expect?

```
❯ devpi-init --help
usage: devpi-init [-h] [-c CONFIGFILE] [--role {master,replica,standalone,auto}] [--master-url MASTER_URL] [--serverdir DIR] [--storage NAME] [--keyfs-cache-size NUM]
                  [--no-root-pypi] [--root-passwd ROOT_PASSWD] [--root-passwd-hash ROOT_PASSWD_HASH]

...
  --no-root-pypi        don't create root/pypi on server initialization. [False]
  --root-passwd ROOT_PASSWD
                        initial password for the root user. This option has no effect if the user 'root' already exist. []
  --root-passwd-hash ROOT_PASSWD_HASH
                        initial password hash for the root user. This option has no effect if the user 'root' already exist. [None]

```

Reply:
🤦 
Yes, we can and should. Good find :)